### ISSUE TITLE
Adding new fields in `PaletteEntry`

### ISSUE THREAD

**Issue by thekingofcity** [wongchito/RailMapGenerator#162]
Current `PaletteEntry` is lacking some features like:

- *express* and *direct* colors for Shanghai Metro line 16
- *pass* color for Shanghai Metro line pujiang
- Image pattern for Chongqing Rail Transit line 3

To add these colors as extensions, `PaletteEntry`'s fields should be enlarged.

**Comment by thekingofcity** [comment]
As there are a lot of colors for different parts, what about merging them into the `color` field? @wongchito 

But then what should we do to the `Theme` and `useLineList` conversion function?

Or simply keep the original layout of `PaletteEntry` and keep adding fields in it?

**Comment by thekingofcity** [comment]
![image](https://user-images.githubusercontent.com/3353040/140518019-d8d2d72b-3a68-4444-b09c-6f8e6c065111.jpeg)

**Comment by wongchito** [comment]
I agree with the merging `colour` field suggestion for Palette. However, I would prefer doing this after we separate the whole palette module from RMG, to avoid introducing massive code change in RMG and also allow us to have an easier management for rmg-palette (including contributions).

**Comment by thekingofcity** [comment]
So it looks like we need another tag like `long-term`. 🤣

**Comment by XHG78999** [comment]
> So it looks like we need another tag like `long-term`. 🤣

“feature”?

**Comment by thekingofcity** [comment]
> “feature”?

Not quite, `long-term` refers to something that will or will not happen in the feature. It may fall into `feature` or `won't fix` in the end.

BTW, this will be fixed very soon in the new railmapgen/rmg-palette.
